src/stpipe/datamodel.py

```diff
@@ -0,0 +1,34 @@
+
+import abc
+
+class DataModel(abc.ABC):
+    """
+    This Abstract Base Class is intended to cover multiple implmentations of
+    data models so that each will be considered an appropriate subclass of this
+    class without requiring that they inherit this class.
+    """
+
+    @classmethod
+    def __subclasshook__(cls, C):
+        """
+        Psuedo subclass check based on these attributes and methods
+        """
+        if cls is DataModel:
+            mro = C.__mro__
+            if (any([hasattr(CC, "crds_observatory") for CC in mro]) and
```

Comment:
What do we want to do about `DataModel.meta.filename`?  AFAICT, that's the only other model attribute accessed by stpipe.  It may be possible to remove references to it but at a glance that doesn't seem trivial.  Maybe we need a new ABC method or property that retrieves the filename?  It doesn't look like stpipe needs to modify it.

Reply:
I wondered about this, but it wasn't obvious to me how to deal with it since `meta` isn't obviously a requirement of `DataModel`, but imposed by a schema. In the end I figured that the requirement would be implicit, but if you have a good way to deal with it, I'm happy to use it. So add a property to all `DataModel`s?

Reply:
I think that would work out nicely, something like:

```
@abc.abstractmethod
def get_filename(self):
  pass
```

Then in JwstDataModel / RomanDataModel that can be implemented like:

```
def get_filename(self):
  return self.meta.filename
```

(I'm not sure about property vs method, but the get_ is nice because it has less of a chance of colliding with a schema property).

Reply:
This would be good, but for the sake of painless transitions, I'd suggest making the changes first for JWST and romancal, then here when they support this method.

Reply:
Even better than a method would be a filename property.  I suspect we will never want to pass an argument to get the filename.

Reply:
Given that we're treating all schema properties as `DataModel` instance attributes, it seems safest to namespace the ABC methods somehow to help prevent collisions.  The method names starting with `get_` are a natural way to do that...

Reply:
(we may want to change `crds_observatory` to a method for the same reason)

Reply:
Added an issue to address this later: https://github.com/spacetelescope/stpipe/issues/21